src/Handlers/SystemWeb/IHttpHandlerFactory.cs

```diff
@@ -0,0 +1,38 @@
+//------------------------------------------------------------------------------
+// <copyright file="IHttpHandlerFactory.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//------------------------------------------------------------------------------
+
+/*
+ * Handler factory interface
+ */
+namespace System.Web {
+    /*
+     * Handler factory -- gets Handler by requestType,path,file
+     */
+
+    /// <devdoc>
+    ///    <para>
+    ///       Defines the contract that factories must implement to dynamically
+    ///       create IHttpHandler instances.
+    ///    </para>
+    /// </devdoc>
+    public interface IHttpHandlerFactory {
```

Comment:
this should be added to dotnet/systemweb-adapters. I'll see if that can be done before we release 2.0

Reply:
so for now, it's fine, but general APIs like this are ok in the systemweb-adapters project